### PR TITLE
RHMAP-15494 - update Hello World cloud app as part of node 6 update

### PR DESCRIPTION
**Pull request by rachael-oregan**
## Motivation
Verify that the Hello World Cloud App could be built with node 6, that the tests passed and that the readme was up to date.

## Result

- Removed the `server.js` file from the acceptance tests. `application.js` can be used instead of this.

- Updated the gruntfile.

- Updated the npm scripts in `package.json` and the `devDependencies` and `dependencies`

- Removed node 0.10 and added node 6 to `travis.yml`

- Updated the readme to reflect the changes.

## Jira
https://issues.jboss.org/browse/RHMAP-15494

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,35 +3,43 @@
...
+    "cors": "~2.8.3",
+    "express": "~4.15.3",
+    "fh-mbaas-api": "~7.0.11",
+    "grunt": "^1.0.1",
+    "request": "2.81.0"
   },
   "devDependencies": {
     "grunt-concurrent": "latest",
-    "grunt-contrib-jshint": "^0.11.3",
+    "grunt-contrib-jshint": "^1.1.0",
```
Comment (sedroche) [inline package.json]: I think we are preferring `~` to `^` now. Can you update these?

File: test/server.js [hunk 2]
```diff
@@ -1,42 +0,0 @@
-var express = require('express');
```
Comment (sedroche) [inline test/server.js]: Why are these tests removed?

Comment (jcesarmobile) [inline test/server.js]: server.js was a fake server used for the acceptance tests, but it's not really needed. We have removed it from other templates too

Comment (sedroche) [inline test/server.js]: :+1:

File: Gruntfile.js [hunk 3]
```diff
@@ -105,10 +106,19 @@ module.exports = function(grunt) {
         },
         command: [
           'rm -rf coverage cov-accept',
-          'env NODE_PATH=. ./node_modules/.bin/istanbul cover --dir cov-accept ./node_modules/.bin/turbo -- --setUp=test/accept/server.js --tearDown=test/accept/server.js test/accept',
+          'env NODE_PATH=. ./node_modules/.bin/istanbul cover --dir cov-accept ./node_modules/.bin/_mocha -- -u exports -R spec ./test/accept/*.js',
           './node_modules/.bin/istanbul report',
           'echo "See html coverage at: `pwd`/coverage/lcov-report/index.html"'
```
Comment (jcesarmobile) [inline Gruntfile.js]: the coverage of accept tests is at cov-accept/lcov-report/index.html, for unit tests is in cov-unit folder, coverage folder has the coverage for both.
You can change the echo test on line 99 to something like
https://github.com/feedhenry-templates/saml-cloud-app/blob/master/Gruntfile.js#L98
and on line 111 to something like https://github.com/feedhenry-templates/saml-cloud-app/blob/master/Gruntfile.js#L110

File: package.json [hunk 4]
```diff
@@ -3,35 +3,43 @@
...
-    "istanbul": "0.2.7",
-    "load-grunt-tasks": "^0.4.0",
-    "mocha": "^2.1.0",
-    "proxyquire": "0.5.3",
-    "should": "2.1.1",
-    "sinon": "^1.17.2",
-    "supertest": "0.8.2",
-    "time-grunt": "^0.3.2"
+    "grunt-plato": "~1.4.0",
+    "grunt-shell": "^2.1.0",
```
Comment (thailekha) [inline package.json]: I'm not sure if the major should be bumped

File: Gruntfile.js (hunk 3 above)
Comment (rachael-oregan) [inline Gruntfile.js]: Nice catch thanks

File: package.json (hunk 1 above)
Comment (rachael-oregan) [inline package.json]: Yep I'll update this, thanks

File: package.json (hunk 4 above)
Comment (rachael-oregan) [inline package.json]: Ya you're right my bad. They won't be backward compatible with node 4. I'll change them back thanks

**Pull request merged**